config/configgrpc/configgrpc.go

```diff
@@ -273,6 +274,22 @@ func (gss *GRPCServerSettings) ToListener() (net.Listener, error) {
 
 // ToServerOption maps configgrpc.GRPCServerSettings to a slice of server options for gRPC.
 func (gss *GRPCServerSettings) ToServerOption(host component.Host, settings component.TelemetrySettings) ([]grpc.ServerOption, error) {
+
+	switch gss.NetAddr.Transport {
+	case "tcp", "tcp4", "tcp6", "udp", "udp4", "udp6":
+		if host, _, err := net.SplitHostPort(gss.NetAddr.Endpoint); err != nil {
+			return nil, fmt.Errorf("failed to parse endpoint: %w", err)
+		} else if ip := net.ParseIP(host); ip != nil && ip.IsUnspecified() {
```

Comment:
No need for else when return. Makes code easier to read.

Reply:
`host` is local to the if block, will define it outside :+1: